src/asInput/index.jsx

```diff
@@ -41,7 +41,7 @@ export const defaultProps = {
   themes: [],
 };
 
-const asInput = (WrappedComponent, labelFirst = true) => {
+const asInput = (WrappedComponent, inputType = undefined, labelFirst = true) => {
```

Comment:
Adding in this `inputType` to help with styling logic. I could see this being leverage for the upcoming Search input box that Dahlia is going to be working on (label layout inline with input field instead of above). Also if any other input types need specific classes/formatting this will help with that logic. If there is a better way to determine this that I was unaware of, let me know! :)

Reply:
This is probably a more flexible approach than the existing `labelFirst` param. I wonder if we should kill `labelFirst` and just lean primarily on `inputType` for label styling logic?

Reply:
I could see an argument for a single input type wanting to be able to support both a label before/after the input field, which is why I didn't simply leverage the labelFirst option for the checks I was doing below. I'm inclined to leave it for now, until we're certain that we won't want it later.